third_party/xla/xla/stream_executor/gpu/gpu_blas_lt_adaptor.h

```diff
@@ -0,0 +1,62 @@
+/* Copyright 2023 The OpenXLA Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef XLA_STREAM_EXECUTOR_GPU_GPU_BLAS_LT_H_
+#define XLA_STREAM_EXECUTOR_GPU_GPU_BLAS_LT_H_
+
+#include <cstdint>
+#include <utility>
+
+#include "xla/stream_executor/blas.h"
+#include "xla/stream_executor/gpu/gpu_blas_lt_gemm_runner.h"
+
+namespace stream_executor::gpu {
+
+template <typename TBlasSupport>
```

Comment:
IIUC, to use this CRTP-based adaptor [here in matmul_op_impl.h](https://github.com/ROCm/tensorflow-upstream/blob/r218-hipblaslt-nonxla-1126/tensorflow/core/kernels/matmul_op_impl.h#L729-L730) we should also need `BaseBlasSupport` and calling like this?
```
  // Specify the template type explicitly
  GpuBlasLtAdaptor<BaseBlasSupport> adaptor;

  // Call the method
  bool success = adaptor.DoBlasGemmBatched();
```

Reply:
Real adaptor would be better than just overriding the concrete function. However there are so many overloads that need to be duplicated just to propagate the call further. I thought about generating them through the template helper recursively but it makes everything complicated. So the simplest solution is to just inherit.

Reply:
Also is this only valid for ROCM? If so we don't even need to make it template. We could directly inherit from ROCMBlas